### PR TITLE
New Caterva plugins example

### PR DESCRIPTION
**Pull request by oscargm98**

### CODE REVIEW

File: examples/example_plugins.c [hunk 1]
```diff
@@ -44,11 +51,11 @@ int main() {
 
     caterva_config_t cfg = CATERVA_CONFIG_DEFAULTS;
     cfg.nthreads = 1;
-    cfg.splitmode = BLOSC_ALWAYS_SPLIT;
-    cfg.compcodec = BLOSC_CODEC_NDLZ;
-    cfg.compmeta = 4;
+    cfg.compcodec = BLOSC_CODEC_NDLZ;       // use the NDLZ codec trough its plugin
+    cfg.splitmode = BLOSC_ALWAYS_SPLIT;     // needed to work correctly with NDLZ codec
+    cfg.compmeta = 4;                       // needed to work correctly with NDLZ codec
```
Comment (FrancescAlted) [inline examples/example_plugins.c]: Can you explain more what the 4 means?  You can open a couple of comment lines for that; this is an example, so don't be afraid of eplaining too much ;-)

Comment (oscargm98) [reply examples/example_plugins.c]: Of course! No probem

File: examples/example_plugins.c [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
+    caterva_config_t cfg = CATERVA_CONFIG_DEFAULTS;
+    cfg.nthreads = 1;
+    cfg.compcodec = BLOSC_CODEC_NDLZ;       // use the NDLZ codec trough its plugin
+    cfg.splitmode = BLOSC_ALWAYS_SPLIT;     // needed to work correctly with NDLZ codec
+    cfg.compmeta = 4;                       // NDLZ metainformation: - it calls the 4x4 version if meta == 4
+                                            //                       - it calls the 8x8 version if meta == 8
+    cfg.complevel = 5;
+                                            // we could use a filter plugin by accessing cfg.filters[]
```
Comment (FrancescAlted) [inline examples/example_plugins.c]: In general, inline comments on the right briefly comments the code on the left.  The case of the line above is a general comment, so better align it to the left.

Also the comment would read better as `// We could use a filter plugin by setting cfg.filters[]`.  Hmmm, pondering about this comment, I suggest to add another example for a plugin filter (e.g. `example_filter_plugin.c`).

Comment (oscargm98) [reply examples/example_plugins.c]: Corrected. I am already developing a new filters example

Comment (FrancescAlted) [reply examples/example_plugins.c]: Cool.  Thanks!

**Pull request merged**